### PR TITLE
Architecture selection options

### PR DESCRIPTION
**Pull request by yuchong-pan**
Added architecture selection options (`--arm32`, `--x86`, `--x64`) for both of the `list` command and the `uninstall` command.

UI issues:
1. For the option combinations e.g. `--all-but-latest --x64 --sdk`, is the "latest" here mean the latest over all SDKs, or the latest over x64 SDKs only?
_Response from @KathleenDollard: The latest over x64 SDKs only._

_Note: This pull request is currently a draft and should be reviewed after Pull Request #12 has been merged into `master`._

### CODE REVIEW

File: src/dotnet-uninstall/Messages.cs [hunk 1]
```diff
@@ -34,8 +34,11 @@ internal static class Messages
         public static readonly string UninstallVerbosityOptionArgumentName = "VERBOSITY-LEVEL";
         public static readonly string UninstallVerbosityOptionArgumentDescription = "The specified MSBuild verbosity level";
 
-        public static readonly string SdkOptionDescription = "Uninstall the .NET Core SDKs only";
-        public static readonly string RuntimeOptionDescription = "Uninstall the .NET Core Runtimes only";
+        public static readonly string SdkOptionDescription = "Uninstall .NET Core SDKs only";
```
Comment (wli3) [inline src/dotnet-uninstall/Messages.cs]: you should start to convert to resx now. The longer you delay that, the harder it gets

File: src/dotnet-uninstall/Shared/BundleInfo/BundleVersion.cs [hunk 2]
```diff
@@ -44,7 +44,7 @@ public static BundleVersion FromInput<TBundleVersion>(string input)
                 return RuntimeVersion.FromInput(input);
             }
 
-            throw new ArgumentOutOfRangeException();
+            throw new ArgumentException();
```
Comment (wli3) [inline src/dotnet-uninstall/Shared/BundleInfo/BundleVersion.cs]: nit: even only the developer should see this exception, it is usually good to add some explaining why. So people can debug faster. You can add something like "cannot find enum matching"

File: test/dotnet-uninstall.Tests/Shared/Filterers/AllBelowOptionFiltererTests.cs [hunk 3]
```diff
@@ -115,15 +122,114 @@ public static IEnumerable<object[]> GetDataForTestFiltererGood()
...
+                },
+                BundleType.Runtime,
+                BundleArch.X64 | BundleArch.Arm32
             };
         }
 
         [Theory]
         [MemberData(nameof(GetDataForTestFiltererGood))]
-        internal void TestAllBelowOptionFiltererGood(IEnumerable<Bundle> testBundles, string argValue, IEnumerable<Bundle> expected, BundleType typeSelection)
+        internal void TestAllBelowOptionFiltererGood(IEnumerable<Bundle> testBundles, string argValue, IEnumerable<Bundle> expected, BundleType typeSelection, BundleArch archSelection)
```
Comment (wli3) [inline test/dotnet-uninstall.Tests/Shared/Filterers/AllBelowOptionFiltererTests.cs]: put expected to the end, everything before that is input. VS has automatic refactoring to change the signature, should be easy to do

File: src/dotnet-uninstall/Messages.cs (hunk 1 above)
Comment (yuchong-pan) [reply src/dotnet-uninstall/Messages.cs]: I agree. But my concern is that there are always pending PRs. I will probably do that during the weekend if we get all merged in today.

File: test/dotnet-uninstall.Tests/Shared/Filterers/AllBelowOptionFiltererTests.cs (hunk 3 above)
Comment (yuchong-pan) [reply test/dotnet-uninstall.Tests/Shared/Filterers/AllBelowOptionFiltererTests.cs]: Discussed offline. This will be too time-consuming for now.

**Pull request merged**